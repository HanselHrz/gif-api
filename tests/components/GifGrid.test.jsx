import { render, screen } from "@testing-library/react";
import { GifGrid } from "../../src/components/GifGrid";
import { useFetchGifs } from "../../src/hooks/useFetchGifs";

jest.mock("../../src/hooks/useFetchGifs");

describe("Pruebas en <GifGrid/>", () => {
  const category = "mr. robot";
  test("Debe de mostrar el loading inicialmente ", () => {
    useFetchGifs.mockReturnValue({
      images: [],
      isLoading: true,
    });
    render(<GifGrid category={category} />);
    expect(screen.getByText("Loading..."));
    screen.debug();
  });

  test("Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs", async () => {
    const gif = [
      {
        id: "ABC",
        title: "Saitama",
        url: "https://localhost/cualquier.jpg",
      },
      {
        id: "123",
        title: "Rick",
        url: "https://localhost/cualquier.jpg",
      },
    ];
    useFetchGifs.mockReturnValue({
      images: gif,
      isLoading: true,
    });
    render(<GifGrid category={category} />);
    expect(screen.getAllByRole("img").length).toBe(gif.length);
  });
});
