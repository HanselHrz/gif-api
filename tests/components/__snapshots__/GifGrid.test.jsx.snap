// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> Debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<div>
  <h3>
    mr. robot
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
